File: 백준/Bronze/10810. 공 넣기/공 넣기.c
#include <stdio.h>
#include <stdlib.h>

int main(void) {
    int N, M = 0;
    scanf("%d %d", &N, &M);
    int* arr = (int*)calloc(N, sizeof(int));
    int i, j, k, index = 0;
    while (index < M) {
        scanf("%d %d %d", &i, &j, &k);
        for(int x = i - 1; x < j; ++x)
			arr[x] = k;
		index++;
    }
    for (int a = 0; a < N; ++a)
        printf("%d ", arr[a]);
    free(arr);
    return 0;
}

File: 백준/Bronze/10810. 공 넣기/공 넣기.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		int n = sc.nextInt();
		int[] arr = new int[n];
		int m = sc.nextInt();
		int i, j, k = 0;
		int index = 0;
		while(true) {
			if(index >= m)
				break;
			i = sc.nextInt();
			j = sc.nextInt();
			k = sc.nextInt();
			for(int x = i - 1; x < j; ++x) {
				arr[x] = k;
			}
			index++;
		}
		
		for(int a : arr)
			System.out.printf("%d ", a);
	}
}